Alchemy/DirectXUtil/CG32bitImage.cpp

```diff
@@ -1038,33 +976,273 @@ void CG32bitImage::DrawDot (int x, int y, CG32bitPixel rgbColor, MarkerTypes iMa
 			//	Draw the top line
 
 			if (yTop >= m_rcClip.top && yTop < m_rcClip.bottom)
-				{
-				CG32bitPixel *pRowStart = GetPixelPos(0, yTop);
-				CG32bitPixel *pPos = pRowStart + Max((int)m_rcClip.left, xLeft);
-				CG32bitPixel *pPosEnd = pRowStart + Min(xRight+1, (int)m_rcClip.right);
+			{
+				CG32bitPixel* pRowStart = GetPixelPos(0, yTop);
+				CG32bitPixel* pPos = pRowStart + Max((int)m_rcClip.left, xLeft);
+				CG32bitPixel* pPosEnd = pRowStart + Min(xRight + 1, (int)m_rcClip.right);
 
 				while (pPos < pPosEnd)
 					*pPos++ = rgbColor;
-				}
+			}
 
 			//	Draw the bottom line
 
 			if (yBottom >= m_rcClip.top && yBottom < m_rcClip.bottom)
+			{
+				CG32bitPixel* pRowStart = GetPixelPos(0, yBottom);
```

Comment:
I think this is obvious, but make sure that you don't write to any pixel outside of the clip rect (m_rcClip). At best you might override something; at worse you'll crash because of access violation.

SetPixel checks for bounds internally, so you don't need to worry. Same with Fill. But if you're going direct with GetPixelPos, make sure you're always in bounds.

Again, I think you're already doing this, but just double-check to make sure.

Reply:
Checked, all uses of *pPos were existing code that was re-organized by shape and the bounds checks for m_rcClip are still in place. The new code is all using SetPixel/Fill.